json/marshallers.go

```diff
@@ -197,3 +198,31 @@ func ToBase64(encoder *jsontext.Encoder, t []byte, options json.Options) error {
 func ToHex(encoder *jsontext.Encoder, t []byte, options json.Options) error {
 	return encoder.WriteToken(jsontext.String(hex.EncodeToString(t)))
 }
+
+func EncodeBase58(bytes []byte) string {
+	return base58.Encode(bytes)
+}
+
+func EncodeBase64(bytes []byte) string {
+	return base64.StdEncoding.EncodeToString(bytes)
+}
+
+func EncodeHex(bytes []byte) string {
+	return hex.EncodeToString(bytes)
+}
+
+// Encode encodes the given bytes using the specified encoding.
+func Encode(bytesEncoding string, bytes []byte) string {
+	switch {
+	case strings.EqualFold(bytesEncoding, "base58"):
```

Comment:
here we could switch based on the length, as mentionned earlier.. just an idea but I think it would make sense